Freshness check dataset per time window (#499)
databrickslabs/dqx: docs/dqx/docs/reference/quality_rules.mdx

> | `is_data_arriving_on_schedule`     | Does freshness check that validates records atleast X number of records arrive at every X minutes window time.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | `column`: column of timestamp column (can be a string column name or a column expression), `window_minutes`: Time interval in minutes to check for data arrival, `min_records_per_interval`: Minimum number of records expected per interval, `lookback_windows`: How many windows to look back from `curr_timestamp`, `row_filter`: Optional SQL expression to filter rows before checking, `curr_timestamp`: Optional current timestamp column. If not provided, current_timestamp() function is used.                                                                                                                                                                                                                                                                                                                                                                                                     |

Reviewer: pls simplify this, remove row_filter as it is standard filter that can be used in all checks, we just is it to push it down
```suggestion
| `is_data_arriving_on_schedule`     | Freshness check that validates whether at least X records arrive within every X-minute time window.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | `column`: timestamp column (can be a string column name or a column expression), `window_minutes`: Time interval in minutes to check for data arrival, `min_records_per_interval`: Minimum number of records expected per interval, `lookback_windows`: How many windows to look back from `curr_timestamp`, `curr_timestamp`: Optional current timestamp column. If not provided, current_timestamp() function is used.                                                                                                                                                                                                                                                                                                                                                                                                     |
```

      criticality="error",
      columns=[["col3", "col5"], ["col1"]]  # apply the check for each list of columns
    ).get_rules(),

> 

Reviewer: please add the example to the yaml examples as well
